### PR TITLE
*: allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond (#29699)

### LINKED ISSUE #29698: Allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond
## Feature Request

**Is your feature request related to a problem? Please describe:**

Allow setting `@@tidb_tso_client_batch_max_wait_time` to a float millisecond. Such as:

```mysql
set global tidb_tso_client_batch_max_wait_time = 0.5
```

### LINKED ISSUE #29722: Value out of the range will only set the session variable leaving PD client unset.
### 1. Minimal reproduce step (Required)

image

### 2. What did you expect to see? (Required)

The PD client sets this truncated value.

### 3. What did you see instead (Required)

The PD client set the original value and failed.

### 4. What is your TiDB version? (Required)

31a4f6abbb2d32460aa844a9641af7c30a7d2cda

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #29699 to release-5.3
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/29751
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/29751:release-5.3-3d9196ddea80
```

---

Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #29698, close #29722.

### What is changed and how it works?

* Allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond.
* Fix the bug that value out of the range will only set the session variable leaving PD client unset.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

image

### Release note

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -65,8 +65,13 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
+<<<<<<< HEAD
 	github.com/tikv/client-go/v2 v2.0.0-alpha.0.20211029104011-2fd3841894de
 	github.com/tikv/pd v1.1.0-beta.0.20211029083450-e65f0c55b6ae
+=======
```
Comment (nolouch) [inline go.mod:71]: some conflicts. cc @JmPotato

**Review by JmPotato — APPROVED** [review]

**Comment by JmPotato** [comment]
@nolouch @morgo PTAL.

**Review by nolouch — APPROVED** [review]

**Pull request merged**